### PR TITLE
Added Dockerfile

### PR DESCRIPTION
**Pull request by marceloalmeida**

### CODE REVIEW

**Comment by marceloalmeida** [comment]
I signed it!

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# build stage
+FROM golang:alpine AS build-env
+ADD . /src
+ENV CGO_ENABLED=0 GOBIN=$GOPATH/bin
+RUN env &&  apk --update add git
```
Comment (stapelberg) [inline Dockerfile]: Do we still need git if we build using Go modules?

Comment (stapelberg) [inline Dockerfile]: Can you remove the double space?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+RUN cd /src && go get ./... && go build -o dnsmasq_exporter
```
Comment (stapelberg) [inline Dockerfile]: Can we use `go install` instead? Alternatively, remove the GOBIN environment variable above :)

File: README.md [hunk 3]
```diff
@@ -31,6 +31,12 @@ systemctl daemon-reload
 systemctl enable --now dnsmasq_exporter
 ```
 
+### Alternative usage
+```shell
+docker build -t dnsmasq_exporter .
+docker run --restart=unless-stopped	--net=host dnsmasq_exporter
```
Comment (stapelberg) [inline README.md]: Can you remove the extra spaces please?

File: Dockerfile (hunk 1 above)
Comment (marceloalmeida) [reply Dockerfile]: Done

File: Dockerfile (hunk 2 above)
Comment (marceloalmeida) [reply Dockerfile]: `go build` with go modules is enough, thanks

File: README.md (hunk 3 above)
Comment (marceloalmeida) [reply README.md]: Done

**Comment by stapelberg** [comment]
Thank you!

**Comment by williamh** [comment]
Hi, it looks like this p/r also added go.mod without go.sum.
To fix this, you should run:

```
$ go mod tidy
```

Also, shouldn't the user referred to in the Dockerfile match the user referred to in the *.service file, and shouldn't that user be something like dnsmasq_exporter instead of prometheus?

Thanks,

William

Also, I have a question about the user referred to in the Dockerfile and the

**Comment by stapelberg** [comment]
> Hi, it looks like this p/r also added go.mod without go.sum.

Fixed in commit fc7234289dff3bb94efa2350356a00acd762a30c, thanks.

> Also, shouldn't the user referred to in the Dockerfile match the user referred to in the *.service file, 

Why? These are 2 separate deployment mechanisms.

> and shouldn't that user be something like dnsmasq_exporter instead of prometheus?

If you want to create that user, feel free to do so. The prometheus user will be present on the system if prometheus is installed, so I just went with that.

**Comment by williamh** [comment]
Does this require prometheus to be on the same system to run?
More importantly, does it require direct access to prometheus in some way?
If not, for security reasons, it is probably better to use a separate user for this exporter.

Thanks,

William

**Comment by stapelberg** [comment]
> Does this require prometheus to be on the same system to run?
> More importantly, does it require direct access to prometheus in some way?

No.

> If not, for security reasons, it is probably better to use a separate user for this exporter.

There’s not a lot of attack surface here. As I mentioned previously, you’re welcome to use a separate user for it. The only reason I chose to run it as the prometheus user is convenience.

**Comment by williamh** [comment]
Can you please tag a new release with these changes?

Thanks much.

**Comment by stapelberg** [comment]
Done.

**Pull request merged**